src/ethernet_interface.cpp

```diff
@@ -419,6 +419,16 @@ bool EthernetInterface::originIsManuallyAssigned(IP::AddressOrigin origin)
     );
 }
 
+inline bool isIPv6(const stdplus::SubnetAny& ip)
+{
+    return std::visit(
+        [](const auto& wrappedAddr) {
+            return std::is_same_v<std::decay_t<decltype(wrappedAddr)>,
+                                  stdplus::In6Addr>;
+        },
+        ip.getAddr());
+}
```

Comment:
Just a suggestion: The below should still be cleaner without a lambda.
```
return std::holds_alternative<stdplus::In6Addr>(ip.getAddr());
```

Reply:
yes, it can be simplified. I removed new function as well.
this can be done without function.